chore(js): restore missing gen1 documents (#7769)
aws-amplify/docs: src/pages/[platform]/build-a-backend/auth/connect-your-frontend/switching-authentication-flows/index.mdx

</InlineFilter>

<InlineFilter filters={["angular", "javascript", "nextjs", "react", "react-native", "vue"]}>

For client side authentication there are three different flows:

1. `USER_SRP_AUTH`: The `USER_SRP_AUTH` flow uses the [SRP protocol (Secure Remote Password)](https://en.wikipedia.org/wiki/Secure_Remote_Password_protocol) where the password never leaves the client and is unknown to the server. This is the recommended flow and is used by default.

2. `USER_PASSWORD_AUTH`: The `USER_PASSWORD_AUTH` flow will send user credentials to the backend without applying SRP encryption. If you want to migrate users to Cognito using the "Migration" trigger and avoid forcing users to reset their passwords, you will need to use this authentication type because the Lambda function invoked by the trigger needs to verify the supplied credentials.

3. `CUSTOM_WITH_SRP` & `CUSTOM_WITHOUT_SRP`: Allows for a series of challenge and response cycles that can be customized to meet different requirements.

The Auth flow can be customized when calling `signIn`, for example:

> ```ts 
> signIn({
>   username,
>   password,

Reviewer: ````suggestion
```ts title="src/main.ts"
await signIn({
	username: "hello@mycompany.com",
  password: "hunter2",
````

## `CUSTOM_WITH_SRP` & `CUSTOM_WITHOUT_SRP` flows

Amazon Cognito user pools supports customizing the authentication flow to enable custom challenge types, 
in addition to a password in order to verify the identity of users. These challenge types may include CAPTCHAs
or dynamic challenge questions. The `CUSTOM_WITH_SRP` flow requires a password when calling `signIn`. Both of 
these flows map to the `CUSTOM_AUTH` flow in Cognito.

To define your challenges for custom authentication flow, you need to implement three Lambda triggers for Amazon Cognito.

<Callout>

For more information about working with Lambda Triggers for custom authentication challenges, please visit [Amazon Cognito Developer Documentation](https://docs.aws.amazon.com/cognito/latest/developerguide/user-pool-lambda-challenge.html).

</Callout>

### Custom authentication flow

To initiate a custom authentication flow in your app, call `signIn` without a password. A custom challenge needs to be answered using the `confirmSignIn` API:

> ```ts title="amplify/auth/handler.ts"
> import { signIn, confirmSignIn } from 'aws-amplify/auth';
> 
> export async function handleSignIn(username: string) {
>   const challengeResponse = 'the answer for the challenge';
>   try {
>     const { nextStep } = await signIn({
>       username,
>       options: {
>         authFlowType: 'CUSTOM_WITHOUT_SRP',
>       },
>     });
>     if (nextStep.signInStep === 'CONFIRM_SIGN_IN_WITH_CUSTOM_CHALLENGE') {
>       try {
>         // to send the answer of the custom challenge
>         const output = await confirmSignIn({ challengeResponse });
>         console.log(output);
>       } catch (err) {
>         console.log(err);
>       }
>     }
>   } catch (err) {
>     console.log(err);
>   }
> }
> 
> ```

Reviewer: ````suggestion
```ts title="src/main.ts"
import { signIn, confirmSignIn } from 'aws-amplify/auth';

const challengeResponse = 'the answer for the challenge';

const { nextStep } = await signIn({
  username,
  options: {
    authFlowType: 'CUSTOM_WITHOUT_SRP',
  },
});

if (nextStep.signInStep === 'CONFIRM_SIGN_IN_WITH_CUSTOM_CHALLENGE') {
  // to send the answer of the custom challenge
  await confirmSignIn({ challengeResponse });
}
```
````

filename for this snippet will be some client-side file, also reduced the snippet to showcase the API

Author: Got it, thank you for correcting it 👍

### CAPTCHA authentication

> To create a CAPTCHA challenge with a Lambda Trigger, please visit [AWS Amplify Google reCAPTCHA challenge example](https://docs.amplify.aws/react/build-a-backend/functions/examples/google-recaptcha-challenge/) for detailed examples. 

Reviewer: ```suggestion
To create a CAPTCHA challenge with a Lambda Trigger, please visit [AWS Amplify Google reCAPTCHA challenge example](/[platform]/build-a-backend/functions/examples/google-recaptcha-challenge/) for detailed examples. 
```
